### PR TITLE
Store solver address with quote

### PR DESCRIPTION
**Pull request by MartinquaXD**
Stores the solver that provided a given quote in preparation for CIP-27.

This change requires us to specify a solver address with a price estimator. I decided to pass a list of arguments delimited with `|` and `,` since this is already common in the code base. Note that this can go away when we switch over to driver price estimators. Also I removed the default arguments for (native) price estimators since it's now unclear which address should be attached to the `Baseline` price estimator.
With the argument parsing in place the PR just has to pipe the address throughout the price estimation and quoting logic.

### Test Plan
Extended unit tests so at least writing to and reading from the DB is covered.

Still unclear how to nicely test that creating an order with an existing quote will correctly copy the solver address of the quote to the `order_quote`. (I'm pretty confident it works but testing this e2e would still be nice)

### CODE REVIEW

**Comment by harisang** [comment]
Referencing issue #1583

File: crates/autopilot/src/arguments.rs [hunk 1]
```diff
@@ -82,13 +82,7 @@ pub struct Arguments {
...
     /// native token.
-    #[clap(
-        long,
-        env,
-        default_value = "Baseline",
-        value_enum,
-        use_value_delimiter = true
-    )]
+    #[clap(long, env, value_enum, use_value_delimiter = true)]
     pub native_price_estimators: Vec<shared::price_estimation::PriceEstimatorType>,
```
Comment (nlordell) [inline crates/autopilot/src/arguments.rs:86]: There shouldn't be solver addresses associated with native price queries. Rewards are just tied to actual quotes and not the native token prices used for the estimates.

Comment (MartinquaXD) [reply crates/autopilot/src/arguments.rs:86]: Rewards are also not tied to limit orders. My plan was to have a consistent data model (i.e. store the address for all orders) and then let analytics simply filter out all orders which should not be rewarded.

Comment (nlordell) [reply crates/autopilot/src/arguments.rs:86]: The consequence of this change is that `cargo run -p orderbook` no longer works by default. You have to remember to specify both `--price-estimator` and `--native-price-estimator`.

I agree with the "My plan was to have a consistent data model" argument, but maybe to make testing easier we can have `Baseline|0x0000000000000000000000000000000000000000` as the default argument value for the flags?

**Review by nlordell — APPROVED** [review]

**Comment by nlordell** [comment]
Note that CI is failing because `cargo +nightly fmt` changed behaviour for `let .. = .. else { .. }` statements.

File: crates/shared/src/price_estimation.rs [hunk 2]
```diff
@@ -41,16 +40,15 @@ use {
...
-    OneInch,
-    Yearn,
-    BalancerSor,
+    Baseline(H160),
+    Paraswap(H160),
+    ZeroEx(H160),
+    Quasimodo(H160),
+    OneInch(H160),
+    Yearn(H160),
+    BalancerSor(H160),
```
Comment (cowfee) [inline crates/shared/src/price_estimation.rs:51]: Every type of estimator will have an address (right?), so maybe this should be split into `PriceEstimator = Address + PriceEsimatorKind`.

**Review by cowfee — APPROVED** [review]

File: crates/shared/src/price_estimation.rs (hunk 2 above)
Comment (cowfee) [reply crates/shared/src/price_estimation.rs:51]: It's also obnoxious to have to match on this value all the time.

Comment (harisang) [reply crates/shared/src/price_estimation.rs:51]: The intention is indeed to only allow solvers to become estimators, and so every estimator will have an address. Note however that this means we should create an account for Quasimodo again, as it is the only "solver" that is actually only used as an estimator. (we had delisted it as a solver when we did the pks rotation).

File: crates/autopilot/src/arguments.rs (hunk 1 above)
Comment (nlordell) [reply crates/autopilot/src/arguments.rs]: Oh nice. I actually like the fix of defaulting to the 0 address.

File: crates/shared/src/price_estimation.rs (hunk 2 above)
Comment (nlordell) [reply crates/shared/src/price_estimation.rs]: Ok, so the `--price-estimators` field will be deprecated at some point.

**Pull request merged**